Error: %02d’ directive output may be truncated writing between 2 and 11 bytes into a region of size between 5 and 12 [-Werror=format-truncation=]

### Expected behavior
gvm-libs compiles.

### Current behavior
gvm-libs do not compile:

```
[ 81%] Building C object nasl/CMakeFiles/openvas_nasl_shared.dir/nasl_var.c.o
[ 83%] Building C object nasl/CMakeFiles/openvas_nasl_shared.dir/nasl_wmi.c.o
[ 84%] Building C object nasl/CMakeFiles/openvas_nasl_shared.dir/nasl_isotime.c.o
/home/xxx/src/gvm/9/gvm-libs/nasl/nasl_isotime.c: In function ‘nasl_isotime_now’:
/home/xxx/src/gvm/9/gvm-libs/nasl/nasl_isotime.c:105:45: error: ‘%02d’ directive output may be truncated writing between 2 and 11 bytes into a region of size between 5 and 12 [-Werror=format-truncation=]
       snprintf (timebuf, ISOTIME_SIZE, "%04d%02d%02dT%02d%02d%02d",
                                             ^~~~
/home/xxx/src/gvm/9/gvm-libs/nasl/nasl_isotime.c:105:40: note: directive argument in the range [-2147483647, 2147483647]
       snprintf (timebuf, ISOTIME_SIZE, "%04d%02d%02dT%02d%02d%02d",
                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/stdio.h:862,
                 from /home/xxx/src/gvm/9/gvm-libs/nasl/nasl_isotime.c:54:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:64:10: note: ‘__builtin___snprintf_chk’ output between 16 and 68 bytes into a destination of size 16
   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        __bos (__s), __fmt, __va_arg_pack ());
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [nasl/CMakeFiles/openvas_nasl_shared.dir/build.make:580: nasl/CMakeFiles/openvas_nasl_shared.dir/nasl_isotime.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:350: nasl/CMakeFiles/openvas_nasl_shared.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
```

### Steps to reproduce
```
git fetch
rm -rf BUILD
mkdir BUILD
cd BUILD
cmake ..
make
```

### GVM versions
```
gsa                                       v7.0.3-57-g293cf3c7f
gvm                                       v7.0.3-103-gce3caed0
gvm-libs                                  v9.0.3-7-gdb1306c1
gvm-tools                                 v1.4.1-239-gbb98ebd
openvas-scanner                           v5.1.3-5-ga7a583a
openvas-smb                               v1.0.4-2-g564d5a0
ospd                                      b989e7f
```

### Environment
**Operating system:**
Debian testing

**Installation method / source:**
git checked out sources

--- a/nasl/nasl_isotime.c
+++ b/nasl/nasl_isotime.c
@@ -102,9 +102,13 @@ epoch2isotime (my_isotime_t timebuf, time_t atime)
       struct tm *tp;
 
       tp = gmtime (&atime);
-      snprintf (timebuf, ISOTIME_SIZE, "%04d%02d%02dT%02d%02d%02d",
+      if (snprintf (timebuf, ISOTIME_SIZE, "%04d%02d%02dT%02d%02d%02d",
                 1900 + tp->tm_year, tp->tm_mon+1, tp->tm_mday,
-                tp->tm_hour, tp->tm_min, tp->tm_sec);
+                tp->tm_hour, tp->tm_min, tp->tm_sec) < 0)
+        {
+          *timebuf = '\0';
+          return;
+        }
     }
 }
 

